mcs/tools/mdoc/Mono.Documentation/monodocer.cs

```diff
@@ -5662,4 +5650,171 @@ protected override StringBuilder AppendNamespace (StringBuilder buf, TypeReferen
 	}
 }
 
+/// <summary>Formats attribute values. Should return true if it is able to format the value.</summary>
+class AttributeValueFormatter {
+	public virtual bool TryFormatValue (object v, TypeReference valueType, out string returnvalue)
+	{
+		if (v == null) {
+			returnvalue = "null";
+			return true;
+		}
+		if (valueType.FullName == "System.Type") {
+			var vTypeRef = v as TypeReference;
+			if (vTypeRef != null) 
+				returnvalue = "typeof(" + NativeTypeManager.GetTranslatedName (vTypeRef) + ")"; // TODO: drop NS handling
+			else
+				returnvalue = "typeof(" + v.ToString () + ")";
+			
+			return true;
+		}
+		if (valueType.FullName == "System.String") {
+			returnvalue = "\"" + v.ToString () + "\"";
+			return true;
+		}
+		if (valueType.FullName == "System.Char") {
+			returnvalue = "'" + v.ToString () + "'";
+			return true;
+		}
+		if (v is Boolean) {
+			returnvalue = (bool)v ? "true" : "false";
+			return true;
+		}
+
+		TypeDefinition valueDef = valueType.Resolve ();
+		if (valueDef == null || !valueDef.IsEnum) {
+			returnvalue = v.ToString ();
+			return true;
+		}
+
+		string typename = MDocUpdater.GetDocTypeFullName (valueType);
+		var values = MDocUpdater.GetEnumerationValues (valueDef);
+		long c = MDocUpdater.ToInt64 (v);
+		if (values.ContainsKey (c)) {
+			returnvalue = typename + "." + values [c];
+			return true;
+		}
+
+		returnvalue = null;
+		return false;
+	}
+}
+
+/// <summary>The final value formatter in the pipeline ... if no other formatter formats the value,
+/// then this one will serve as the default implementation.</summary>
+class DefaultAttributeValueFormatter : AttributeValueFormatter {
+	public override bool TryFormatValue (object v, TypeReference valueType, out string returnvalue)
+	{
+		returnvalue = "(" + MDocUpdater.GetDocTypeFullName (valueType) + ") " + v.ToString ();
+		return true;
+	}
+}
+
+/// <summary>Flags enum formatter that assumes powers of two values.</summary>
+/// <remarks>As described here: https://msdn.microsoft.com/en-us/library/vstudio/ms229062(v=vs.100).aspx</remarks>
+class StandardFlagsEnumFormatter : AttributeValueFormatter {
+	public override bool TryFormatValue (object v, TypeReference valueType, out string returnvalue)
+	{
+		TypeDefinition valueDef = valueType.Resolve ();
+		if (valueDef.CustomAttributes.Any (ca => ca.AttributeType.FullName == "System.FlagsAttribute")) {
+
+			string typename = MDocUpdater.GetDocTypeFullName (valueType);
+			var values = MDocUpdater.GetEnumerationValues (valueDef);
+			long c = MDocUpdater.ToInt64 (v);
+			if (c == 0 && values.Keys.Contains (c)) {
+				returnvalue = typename + "." + values [c]; 
+			} else {
+				returnvalue = string.Join (" | ",
+					(from i in values.Keys
+					where (c & i) == i && i != 0
+					select typename + "." + values [i])
```

Comment:
Indentation: prefix `where`, `select` with a space:

    (from ...
     where ...
     select ... )

Alternatively, indent `where`, `select` by another tab level.

The intent is to not "hide" the parenthesis. Having `where` at the same indentation level as `(from` makes them look like peers (unless you read closely), e.g. for them to be peers it *should* be:

    (from  ... )
    where ...

(I can't think of any way for the above to be possible, but regardless...)